tests/model_fields/tests.py

```diff
@@ -60,6 +66,207 @@ def test_field_str(self):
+        self.assertNotIn('readonly_int2', query['sql'])
+        self.assertEqual(inst.readonly_int1, 1)
+        self.assertEqual(inst.readonly_int2, 2)
+
+    def test_bulk_create(self):
+        with CaptureQueriesContext(connection=connection) as capture:
+            ModelWithReadonlyField.objects.bulk_create([
+                ModelWithReadonlyField(
+                    not_readonly='not_readonly',
+                    readonly_int1=1,
+                ),
+                ModelWithReadonlyField(
+                    not_readonly='rabble rabble',
+                    readonly_int2=9001,
+                )
+                ])
+
+        self.assertEqual(len(capture.captured_queries), 1)
+        query = capture.captured_queries[0]
+
+        self.assertIn('not_readonly', query['sql'])
+        self.assertNotIn('readonly_int1', query['sql'])
+        self.assertNotIn('readonly_int2', query['sql'])
+
+        # bulk_create won't fetch the database computed
+        # values so it's no use trying to check it.
+
+    def test_deserialize(self):
+        try:
+            pk = ModelWithReadonlyField.objects.latest('id').id + 1
+        except ModelWithReadonlyField.DoesNotExist:
+            pk = 1
+
+        json_inst = json.dumps(
+            [{
+                'model': 'model_fields.modelwithreadonlyfield',
+                'pk': pk,
+                'fields': {
+                    'not_readonly': 'All your base',
+                    'readonly_int1': 12,
+                },
+            }]
+        )
+
+        # Check that the save works
+        deserialized = serializers.deserialize('json', json_inst)
+        inst = next(deserialized)
+        with CaptureQueriesContext(connection=connection) as capture:
+            inst.save()
+
+        # Because a pk is specified, Django will try to do an UPDATE and then
+        # an INSERT when the UPDATE returns with 0 rows affected
+        self.assertEqual(len(capture.captured_queries), 3)
+        query_1 = capture.captured_queries[0]['sql']
+
+        self.assertTrue(query_1.startswith('UPDATE'), '{} is not UPDATE'.format(query_1))
+        self.assertIn('not_readonly', query_1)
+        self.assertNotIn('readonly_int1', query_1)
+        self.assertNotIn('readonly_int2', query_1)
+
+        query_2 = capture.captured_queries[1]['sql']
+        self.assertTrue(query_2.startswith('INSERT'), '{} is not INSERT'.format(query_2))
+        self.assertIn('not_readonly', query_2)
+        self.assertNotIn('readonly_int1', query_2)
+        self.assertNotIn('readonly_int2', query_2)
+
+
+@unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL-specific tests")
+class PostgreSQLReadonlyTests(ReadonlyTestsMixin, TestCase):
+
+    @classmethod
+    def populate_database_fields(cls):
+        connection.cursor().execute(
+            '''ALTER TABLE "model_fields_modelwithreadonlyfield" '''
+            '''ALTER COLUMN "readonly_int1" SET DEFAULT 1 ''')
```

Comment:
SQLite cannot add a default after the table is created (to my knowledge).

Mysql is clearly something we forgot ^^